keel-titus-plugin/keel-titus-plugin.gradle.kts

```diff
@@ -5,7 +5,7 @@ plugins {
 
 dependencies {
   api(project(":keel-api"))
-  api(project(":keel-ec2-api"))
+  api(project(":keel-titus-api"))
   implementation(project(":keel-core")) // TODO: ideally not
```

Comment:
FWIW, I added a `KeelReadOnlyRepository` to `keel-api` recently, specifically to support plugins. Between that, `TaskLauncher` and `EventPublisher`, I _think_ you should be able to access all the core functionality you need.